### DISCUSSION TITLE
Why does it run the build step again without any changes?

### DISCUSSION THREAD

**Discussion by matthiasgeihs** [risc0/risc0#1528] [Q&A] [answered]
When playing with the examples, I noticed that the compiler seems to run the build steps every time even without any changes.

For example, for the `ecdsa` example it always takes about 50s to compile, even if I simply run the command twice right after each other.

```
ecdsa % cargo run --release -F metal
...
   Compiling ecdsa-example v0.1.0 (.../risc0/examples/ecdsa)
    Finished release [optimized + debuginfo] target(s) in 56.03s
...
ecdsa % cargo run --release -F metal
...
   Compiling ecdsa-example v0.1.0 (.../risc0/examples/ecdsa)
    Finished release [optimized + debuginfo] target(s) in 55.19s
...
```

Is that expected? Why is that?

**Comment by nuke-web3** [comment, marked as answer]
We are deprecating discussions on Github, and encouraging everyone to [join us on discord](https://discord.com/invite/risczero) where we can chat and provide support.

### [Join our Community Discord for dev chit-chat & our support forum](https://discord.com/invite/risczero)

For general questions and problems you are facing with RISC Zero, please **search** our `🙋｜dev-questions-support-forum` first for solutions, by keyword and tags. If you discover no solution, please then ask and questions in our community! We highly encourage everyone also share their understanding by answering questions for others!

Please do introduce yourself on our discord and send me a friend request (`@nuke_web3`) reminding me of this request so we can open up a forum post on it if there isn't a related one already there :grin:

**Reply by nuke-web3** [reply]
I see you did bring this back up here: https://discord.com/channels/953703904086994974/975108547589324910/1281378245568368651
Please open a new discord support forum post as I mentioned if you need more support :pray:
